src/DebugPoints/DebugPointManager.class.st

```diff
@@ -0,0 +1,374 @@
+"
+I am responsible for managing the debug points.
+
+An exception is the removal of Debug points, which is handled in the ""remove"" method of DebugPoint instances.
+"
+Class {
+	#name : 'DebugPointManager',
+	#superclass : 'Object',
+	#category : 'DebugPoints-Base',
+	#package : 'DebugPoints',
+	#tag : 'Base'
+}
+
+{ #category : 'announcement handling' }
+DebugPointManager class >> handleClassRemoved: anAnnouncement [
+
+	DebugPoint all copy do: [ :debugPoint |
+		debugPoint link methods
+			detect: [ :m | m methodClass = anAnnouncement classRemoved ]
+			ifFound: [ debugPoint remove ] ]
+]
```

Comment:
For later: because the DebugPoint model the breakpoints at a higher level than metalinks, it would simplify things (like that method) to know more of the underlying model, for instance here it would make sense that each debug points knows the structural elements on which it is installed/active.

In this case the debug point would know the methods it targets.